examples/client/tui/commands.nim

```diff
@@ -722,6 +722,43 @@ proc command*(self: Tui, command: SendMessage) {.async, gcsafe, nimcall.} =
   else:
     asyncSpawn self.client.sendMessage(command.message)
 
+# Call ----------------------------------------------------------------
+
+proc help*(T: type CallRpc): HelpText =
+  let command = "call"
+  HelpText(command: command, aliases: aliased[command], parameters: @[
+    CommandParameter(name: "method", description: "RPC method"),
+    CommandParameter(name: "params", description: "Array of parameters")
+    ], description: "Calls an Ethereum RPC method")
+
+proc new*(T: type CallRpc, args: varargs[string]): T =
+  var rpcMethod = if args.len > 0: args[0] else: ""
+  var params = newJArray()
+  if args.len > 1:
+    try:
+      params = args[1].parseJson()
```

Comment:
The user might have typed in valid json that contained spaces, e.g. between `"key":` and the following value.

So here I think `args[1..^1]` should be joined on single-space (`" "`) and then parsed.

Also, I think we should have an `else` case for `if args.len > 1` where params is set to the empty string.

Then, in `proc command` below for `command: CallRpc`, the situation re: blank params and invalid JSON can be separated.

Reply:
> `else` case for if `args.len > 1` where params is set to the empty string.

Whoops, looking over it again I see params needs to be of type `JsonNode`. So there's no need for an `else` case, but we could still separate the cases w.r.t. checking if it's JNull else checking the `len` of JArray being `> 0`.